sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -2363,8 +2363,8 @@ class Dataset[T] private[sql](
    * @since 2.2.0
    */
   @DeveloperApi
-  def collectAsArrow(): ArrowRecordBatch = {
-    val allocator = new RootAllocator(Long.MaxValue)
+  def collectAsArrow(
+      allocator: RootAllocator = new RootAllocator(Long.MaxValue)): ArrowRecordBatch = {
```

Comment:
I see this is done so the test can use the same root allocator, would it fail if the tests made a different instance?  I'm just wondering about the broader usage in Spark, like should Spark manage a single root allocator or just create one each operation and allow the user to override like this?